ambry-api/src/main/java/com.github.ambry/config/StoreConfig.java

```diff
@@ -112,6 +119,8 @@ public StoreConfig(VerifiableProperties verifiableProperties) {
     storeEnableHardDelete = verifiableProperties.getBoolean("store.enable.hard.delete", false);
     storeSegmentSizeInBytes =
         verifiableProperties.getLongInRange("store.segment.size.in.bytes", Long.MAX_VALUE, 1, Long.MAX_VALUE);
+    storeHardDeleterSleepTimeWhenCaughtUpInMs =
```

Comment:
This is bit complex, but let me try to explain. Once pause() has been invoked, we need to ensure that the hardDeleter thread is in WAITING state in tests.(check for waitUntilExpectedState() in IndexTest()). What the method does is, check for hard deleter thread's state to become WAITING upto some max time, say X. X is determined based on storeHardDeleterSleepTimeWhenCaughtUpInMs assuming the hard delete thread might be in blocked_waiting state when it was in timed await when caught up. So, if I remove the signalling of "pauseCondition" within pause() method, then we have to wait until storeHardDeleterSleepTimeWhenCaughtUpInMs + few ms to ensure that hard deleter thread would have moved to WAITING state incase it was in timed wait. So, we have two options here.

1. Don't remove the config. We can remove signaling the pauseCondition within pause(). But this also means that there will be a increase in running time of tests, since we wait until the thread moves to WAITING state.
2. Remove the config(which means that value will be set to 10 secs as hard coded in HardDeleter). But have to signal the pauseCondition within pause(). If not, we have to literally wait for 10 secs + few ms to ensure that the thread moves to WAITING state.    

Hope this is clear. If not, we can sync up offline